tools/test_utils.py

```diff
@@ -575,6 +575,162 @@ def run_resnet50_forward_pass_from_artifacts(ngraph_tf_src_dir, artifact_dir,
+            shell=True,
+            stdout=subprocess.PIPE)
+        if res.stdout.decode('utf-8').rstrip() == 0:
+            raise Exception("Can't add 'import ngraph_bridge' to script file: "
+                            + script_file)
+
+    # Setup the env flags
+    import psutil
+    num_cores = int(psutil.cpu_count(logical=False))
+    print("OMP_NUM_THREADS: %s " % str(num_cores))
+    os.environ['OMP_NUM_THREADS'] = str(num_cores)
+    os.environ["KMP_AFFINITY"] = 'granularity=fine,compact,1,0'
+
+    # Delete older logs
+    log_dir = IntelAIModels_dir + '/benchmarks/common/tensorflow/logs'
+    if os.path.exists(log_dir):
+        #os.rmdir(log_dir)
+        filelist = glob.glob(os.path.join(log_dir, "*.log"))
+        for f in filelist:
+            os.remove(f)
+
+    os.chdir(IntelAIModels_dir + '/benchmarks/')
+    # python launch_benchmark.py  --in-graph resnet50_v1.pb  --model-name resnet50v1_5
+    # --framework tensorflow  --precision fp32  --mode inference  --batch-size=1  --socket-id=0
+    cmd = [
+        'python',
+        'launch_benchmark.py',
+        '--in-graph',
+        pretrained_model,
+        '--model-name resnet50v1_5',
+        '--framework tensorflow',
+        '--precision fp32',
+        '--mode inference',
+        '--batch-size=1',
+        '--socket-id=0',
+    ]
+    command_executor(cmd, verbose=True)
+
+    os.chdir(root_pwd)
+
+
+def run_simple_resnet50_infer_from_artifacts(ngraph_tf_src_dir, artifact_dir,
+                                             batch_size, iterations):
+    root_pwd = os.getcwd(
+    )  # e.g. /localdisk/buildkite-agent/builds/aipg-ra-skx-168-2/ngraph/ngtf-cpu-ubuntu
+    artifact_dir = os.path.abspath(artifact_dir)
+    if not os.path.exists(artifact_dir):
+        raise Exception("Can't find artifact dir: " + artifact_dir)
+    ngraph_tf_src_dir = os.path.abspath(ngraph_tf_src_dir)
+    if (len(glob.glob(artifact_dir + "/ngraph_tensorflow_bridge-*.whl")) == 0):
+        install_ngraph_bridge(artifact_dir)
+
+    # Check/download pretrained model
+    pretrained_models_dir = root_pwd + '/pretrained_models'
+    if not os.path.exists(pretrained_models_dir):
+        os.mkdir(pretrained_models_dir, 0o755)
+    os.chdir(pretrained_models_dir)
+    pretrained_model = pretrained_models_dir + '/resnet50_v1.pb'
+    if not os.path.exists(pretrained_model):
+        # wget https://zenodo.org/record/2535873/files/resnet50_v1.pb
+        command_executor(
+            ['wget', 'https://zenodo.org/record/2535873/files/resnet50_v1.pb'],
+            verbose=True)
+        if not os.path.exists(pretrained_model):
+            raise Exception("Can't download pretrained model: " +
+                            pretrained_model)
+    else:
+        print("Using existing pretrained model file: " + pretrained_model)
+
+    os.chdir(root_pwd)
+    cmd = [
+        'python',
+        ngraph_tf_src_dir + '/test/python/test_rn50_infer.py',
+        '--input-graph',
+        pretrained_model,
```

Comment:
Pass `batch_size` to the script here. The script doesn't take `iterations` but takes the number of (synthetic) images to process.